torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -514,14 +514,13 @@ class ShapePropagator {
         return;
       }
       case prim::unchecked_unwrap_optional: {
-	// we know we cannot have None as input, so we can always pass
-	// on the type.
-	if(auto ot = node->input()->type()->cast<OptionalType>()) {
-	  node->output()->setType(ot->getElementType());
-	} else {
-	  node->output()->setType(node->input()->type());
-	}
-	return;
+        // If we have None as input, we need to leave the output type alone
+        if (auto ot = node->input()->type()->cast<OptionalType>()) {
+          node->output()->setType(ot->getElementType());
+        } else if (!node->input()->type()->isSubtypeOf(NoneType::get())) {
+          node->output()->setType(node->input()->type());
+        }
+        return;
```

Comment:
Is this just formatting? Can you please avoid doing formatting only changes?

Reply:
No it's not. I check for NoneType now to avoid propagating a None outside.
I try to. Here I actually messed up the formatting before and just re-formatting the changed line would have made things worse.